### ISSUE TITLE
Group members share the same nickname color, very often

### ISSUE THREAD

**Issue by Vlandmist** [telegramdesktop/tdesktop#2204]
https://s31.postimg.org/50rwqemh7/name_color.png

### Steps to reproduce
1. Open a group chat with multiple contacts
2. Chat
3. Look at the members nickname color in chat

### Expected behaviour
Every name should be clearly identified by a different color. Substantial differences (red, green, pink, blue, black etc etc) not just small gradation changes.

### Actual behaviour
Even in a small group with 5-6 people I always see ppl sharing the same color or at least a very very similar color. It doesn't help the conversation flow

### Configuration
**Operating system:**
Win10

**Version of Telegram Desktop:**
v 0.9.56 26.06.16

**Comment by gerbenvandijk** [comment]
There are more inconsistencies regarding this:

- it is not consistent between platforms. e.g. user A could be blue on the desktop client but red on the iOS client
- it is not consistent between groups. e.g. user A could be blue in chat B but the same user could be orange in chat C

All of this results in mass confusion sometimes; as people think person A is speaking while this is not the case. One of my biggest annoyances on Telegram 😁

**Comment by john-preston** [comment]
@gerbenvandijk Are you sure about different chats? Currently user's color is his own, shared across all the chats this user is in. At least in Telegram Desktop.

**Comment by gerbenvandijk** [comment]
You are right @john-preston - i've just checked this and in all group chats the user indeed has the same color. In a private chat not though (just blue there). The difference between desktop and other platforms is there though, but more marginal as i thought earlier; on iOS users at least have a similar color but just a different tone. 

I'm kind of surprised by the above as a week ago I had a discussion amongst friends on telegram about this issue and then both issues were there, hence my reply here. I'll check with them if it's maybe the android version.

**Comment by maorefaeli** [comment]
It's happening both in android and windows PC versions
Indeed very disappointing, especially since it's open for an year :) 
(I have the same color also on 4 people group - only 2 colors are used instead of 3 and they are orange and light red)

**Comment by rvilla87** [comment]
Any news regarding this issue?

Can we change the member's nickname color?

Thanks

**Comment by krocheah** [comment]
in our five ppl group 2ppl is red and 1 person is reddish pink which one is so close to red. maybe we should change ppl color with a color wheel or something.

**Comment by tobold** [comment]
+1

**Comment by Ravor9** [comment]
It's really frustrating. Could be more clean without profile pic (for non super group) and with more color options. And for super groups with few users colors could fallow the same pattern.

Would be awesome an option to chose your hex color

**Comment by Crono30067** [comment]
Agreed with Ravor9, if a personal hex color option was supported, there would be a much greater variance in colors for group chats. It would also open up more random options for defaults.

**Comment by misterbykl** [comment]
+1

**Comment by misterbykl** [comment]
It is frustrating indeed. Especially for the groups with 10+ 15+ people in. 
Should be resolved asap. Looking forward to it!

**Comment by stek29** [comment]
User is expected to have same color in all chats, so it's kinda a feature

**Comment by NotStarRift** [comment]
Any updates regarding this?

**Comment by Luckz** [comment]
I'm on Desktop and both my friends are orange. Not fun.

**Comment by Memou** [comment]
I also have two orange friends.They are ok with it,but the rest is not.

**Comment by ammaratef45** [comment]
All of my chats are just green and blue

**Comment by aziad1998** [comment]
It's been 3 years since this issue was opened
I have a group and literally everyone is the same colour, that so confusing

**Comment by redphoenix1022** [comment]
Bump

**Comment by iNeonFox** [comment]
Same issue here: we have a chat with 30 people in it, and we all have about 5-6 same colors.

**Comment by Aokromes** [comment]
maybe even https://github.com/telegramdesktop/tdesktop/issues/430#issuecomment-307671466

**Comment by KouriiRaiko** [comment]
I'd really like this issue to stay open, it is definitely an important feature that Telegram is lacking.

**Comment by gerbenvandijk** [comment]
Seconded, its difficult to scan now when a lot of messages have flown by.

**Comment by LazieKat** [comment]
This issue stopped in my groups on its own, but I think it was kinda random. If someone can verify that it was actually "solved" that'd be cool.

**Comment by KoryHunter37** [comment]
I operate a group chat with thousands of users and this issue is always prevelent. There's so many people with the same name color, and also it seems like there's a disproportionate color distribution. (Lots of blues and greens, very few other colors)

**Comment by torchu** [comment]
Yep, this is kinda annoying and I think the solution can't be that hard to implement

**Comment by ItIsApachee** [comment]
It's not related to the issue, but can somebody link me to the part of the code which generates color for a user?

**Comment by bvnp43** [comment]
pls add ability to select color for account other than default that randomly was assigned, its should not be hard to implement

**Comment by Aokromes** [comment]
> pls add ability to select color for account other than default that randomly was assigned, its should not be hard to implement

this is already a thing on 4.12.2
